tests/integration/pilot/mcp/main_test.go

```diff
@@ -78,10 +78,11 @@ func setupConfig(cfg *istio.Config) {
 	cfg.Values["pilot.configSource.subscribedResources[0]"] = "SERVICE_REGISTRY"
 	// ICP doesn't support set with list, so need to override here
 	cfg.ControlPlaneValues = `
-galley:
-  enableServiceDiscovery: true
-pilot:
-  configSource:
-    subscribedResources: ["SERVICE_REGISTRY"]
+values:
```

Comment:
This is not trivial... Was the code incorrect?

Reply:
The rest of changes indicate this is intended

Reply:
Yes, we had to make the config more flexible for other tests to pass